### PR TITLE
add Matthews corrcoef

### LINKED ISSUE #59: Cohen Kappa Score and Matthews Correlation Coefficient Metrics
## 🚀 Feature
I would like to request the (re-) implementation of the Cohen Kappa score and the new implementation of the Matthews Correlation Coefficient (MCC) in PyTorch Lightning's metrics.

### Motivation

The Cohen Kappa and MCC are often used metrics in classification tasks, especially in a medical setting to determine such things as inter-grader reliability. The Kappa score was originally implemented in PyTorch Lightning 0.9 but has disappeared for some reason. The MCC is often seen as the best metric to use in highly imbalanced datasets. The addition of these two metrics would make it more convenient to use PyTorch Lightning for medical tasks and other tasks that involve ground truth uncertainty and imbalanced data.

### Pitch

Implementation of the Cohen Kappa and MCC as metrics in PyTorch Lightning. Both metrics are already available in sci-kit learn.

### Alternatives

Cannot think of any.

### Additional context

None.

### PR DESCRIPTION
**Pull request by SkafteNicki**
# Before submitting

- [x] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure to update the docs?   
- [x] Did you write any new necessary tests?  

## What does this PR do?
Fixes #59 
Second metric to be requested in the issue, Matthews correlation coefficient. Again, just a port of sklearn implementation.

## PR review    
Anyone in the community is free to review the PR once the tests have passed.     
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -15,6 +15,8 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Added `CohenKappa` metric ([#69](https://github.com/PyTorchLightning/metrics/pull/69))
 
 
+- Added `MatthewsCorrcoef` metric ([]())
```
Comment (SkafteNicki) [inline CHANGELOG.md:18]:
```suggestion
- Added `MatthewsCorrcoef` metric ([#98](https://github.com/PyTorchLightning/metrics/pull/98))
```

**Review by Borda — APPROVED** [review]

**Review by justusschock — APPROVED** [review]
